Basic Algorithms/GCD(Euclidean Algorithm)

```diff
@@ -0,0 +1,52 @@
+/*
+	GCD stands for greatest common divisor.
+	Here we're using Euclidean algorithm to find GCD.
+	This alorithm states that GCD(a,b) = GCD(b%a,a)
+	So we created a function that calls another recursive function
+	which works as the same.
+	inside findGCD(int num1, int num2) we keep on calling findGCD(num2%num1,num1)
+	until we have num1 as 0
+	as soon as we get 0 in num1 we're returning num2
+	
+	Time complexity = O(log x)  where x = min(num1,num2) 
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+class GCD {
```

Comment:
No need to declare a class for GCD, simply declare the function GCD and call it recursively with current edge cases and considering the complexities.